code/string-calculator/src/main/java/ru/unn/agile/calc/model/StringCalc.java

```diff
@@ -9,36 +9,44 @@ public StringCalc() {
     }
 
     public double result(final String string) {
+        if (string == null) {
+            throw new IllegalArgumentException("String can't be NULL");
+        }
+        if (string.equals("")) {
+            throw new IllegalArgumentException("String can't be empty");
+        }
+        char firstSymbol = string.charAt(0);
+        String[] parsedNumbers;
+        String[] parsedOperators;
         try {
-            char firstSymbol = string.charAt(0);
-            String[] parsedNumbers = string.split("\\+|\\-|\\*|\\/");
-            String[] parsedOperators = string.split("[0-9\\.]*");
-            int index = 0;
-            char subSymbol = Operation.SUBTRACT.value().charAt(0);
-            if (firstSymbol == subSymbol) {
-                index++;
-                parsedNumbers[index] = subSymbol + parsedNumbers[index];
-                parsedOperators[0] = "";
-            }
+            parsedNumbers = string.split("\\+|\\-|\\*|\\/");
+            parsedOperators = string.split("[0-9\\.]*");
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("String can't contain letters");
```

Comment:
Please change. You catch the NumberFormat Exception but rise the IlligalArgument. Should be used NumberFormat :)

Reply:
Already fixed.